vtex/mkp-app-template
# Create sales channel automatically at save for new sellers

[**Jira Issue - CIT-457**](https://vtex-dev.atlassian.net/browse/CIT-457)

#### What is the purpose of this pull request?

- Add data and inputs to select `country`, `culture info`, `currency` and `timezone` of the sales channel.
- Add mutation and client to create sales channel in catalog.
- Change sales channel select component, since now the seller doesn't need to select which SC to use for the integration anymore.

#### What problem is this solving?

- Sellers can only create up to 3 sales channels by themselves in their store admin. The only way to bypass this rule is to create them via API, so that means that they have to open a ticket to our support to do so. The idea of this feature is to automatically create a sales channel for the marketplace integrations using the App Template, so this manual action won't be needed anymore.
- Each integration will have it's own sales channel, and initially we created it in the `inactive` status, so the seller can review/edit its information before enabling it, or other services (such as the new onboarding experience) can enable it later on.

#### How should this be manually tested?

- Go to this workspace: https://luiza--grocery1.myvtex.com/admin/vtex/external-mkp and fill in the required info and save. A sales channel should be created, along with an affiliate pointing to it.

#### Screenshots or example usage

![2022-06-21 19_10_11-Settings](https://user-images.githubusercontent.com/32556748/175039385-792dd528-2a8c-447a-94b5-f63d5464a0eb.png)
![2022-06-21 19_15_01-Settings](https://user-images.githubusercontent.com/32556748/175039395-c286f61b-9339-4a68-b183-353264bd28d8.png)

#### Types of changes

- [ ] Refactor (doesn't change the behaviour but improves code readability)
- [ ] Infrastructure change (deployment settings and such)
- [ ] Security issue (non-breaking change that fixes a security issue)

#trivial
